### DISCUSSION TITLE
How to handle large excel sheet with 1000+ rows and 20cols? (file size 2MB)

### DISCUSSION THREAD

**Discussion by Heshan969** [PHPOffice/PhpSpreadsheet#4166] [Q&A] [answered]
$spreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load($excelFilePath);

error in my WP site: 

Fatal error: Out of memory (allocated 398458880) (tried to allocate 4096 bytes) in /home/xxxx/public_html/wp-content/plugins/cbxphpspreadsheet/lib/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Collection/Cells.php on line 363
There has been a critical error on this website. Please check your site admin email inbox for instructions.

My RAM : 6GB

**Comment by oleibman** [comment, marked as answer]
Without seeing your spreadsheet, we can't be sure. However, some things which might help are setting your reader to ignore empty cells, ignore rows with no cells, and, if you are not interested in the cell styles, loading data only.
```php
$reader->setReadEmptyCells(false);
$reader->setIgnoreRowsWithNoCells(true);
$reader->setReadDataOnly(true);
```
